### ISSUE TITLE
C API references non-existing "string_cache.h"

### ISSUE THREAD

**Issue by RReverser** [servo/html5ever#203]
`html5ever.h` [references](https://github.com/servo/html5ever/blob/master/capi/include/html5ever.h#L15) `string_cache.h` although that definition file can't be found neither in this repo nor in https://github.com/servo/string-cache.

Also, https://github.com/servo/html5ever/blob/master/examples/capi/tokenize.c example uses `struct h5e_buf` in place of `scache_atom`s in `html5ever.h` and undefined function `h5e_buf_from_cstr`, so I suspect header is just outdated and needs to define new structure instead of including `string_cache.h`.

**UPD**: Actually, if I understand https://github.com/servo/html5ever/blob/master/capi/src/tokenizer.rs#L121 correctly, those functions accept just Tendrils (instead of `scache_atom` / `h5e_buf`), so it should be possible to change definitions like following:

```c
void (*do_start_tag)(void *user, tendril name, int self_closing, size_t num_attrs);
void (*do_tag_attr)(void *user, tendril name, tendril value);
void (*do_end_tag)(void *user, tendril name);
```

and then code like this would work:

```c
tendril buf = TENDRIL_INIT;
tendril_push_buffer(&buf, html, size);
h5e_tokenizer_feed(tok, buf);
```

(although I might be wrong)

**Comment by Ygg01** [comment]
kmcallister (kmc) worked on it, before they left. It was here at  https://github.com/servo/string-cache/pull/84 but was never merged with string-cache master. It has bitroted somewhat (May 2015). 

I'm willing to work on it, tomorrow, since it's one of the rare days I have time to commit to it.

**Comment by asajeffrey** [comment]
Note #204

**Comment by RReverser** [comment]
@Ygg01 Sure, no rush. I'm actually already trying to fix it and come up with PR (no promises yet though).

**Comment by RReverser** [comment]
@asajeffrey I'm confused as to why it's being removed :/ Hoped for the opposite (extending).

**Comment by Ygg01** [comment]
@RReverser If push comes to shove, just copy the capi into a separate crate and publish it independently. You'll probably need to something similar to string-cache.

**Comment by RReverser** [comment]
@Ygg01 Actually that PR to string-cache looks good to me. Need to try as to whether it still works.

**Comment by SimonSapin** [comment]
I don’t think there’s anything worth salvaging in `html5ever.h`. Closing in favor of #204, please keep the discussion there.

**Comment by RReverser** [comment]
@SimonSapin That PR removes C API completely, which might be not the best place for discussing how to fix it :)

**Comment by SimonSapin** [comment]
As maintainer I declare that there will be no fixing this API in this repository. What’s being removed was a proof of concept and was never very useful in the first place as it only exposed the tokenizer. What’s left to discuss is starting points if you or someone else wants to build something else.

**Comment by RReverser** [comment]
That's pretty sad to hear. Thanks for the information.

**Comment by RReverser** [comment]
@SimonSapin Sounds like now all the related issues can be closed altogether :/ #192 #143 #131 #44 #42 #33 #32

**Comment by SimonSapin** [comment]
Indeed. Thanks.
